### DISCUSSION TITLE
Desktop shortcut Ubuntu

### DISCUSSION THREAD

**Discussion by mistamichal** [comfyanonymous/ComfyUI#2397] [General]
Can someone please tell me who to create a desktop shortcut to Comfyui in Ubuntu? I made on for Automatic1111 using the launch.sh script and thought if I made a launch.sh for comfyui, I could just do the same thing but I can't get it to run.
Cheers

**Comment by ltdrdata** [comment]
If your venv is located in the `ComfyUI/venv` path and you plan to place the `launch.sh` in `ComfyUI`, then it would be as follows:

```
#!/bin/bash
source ./venv/bin/activate
python main.py
```

If you want to open browser automatically... add `--windows-standalone-build`

```
#!/bin/bash
source ./venv/bin/activate --windows-standalone-build
python main.py
```

**Reply by mistamichal** [reply]
I don't have a venv. I installed in linux using the manual install for AMD instructions.

Oh, and I should say - my launch.sh is working fine from the terminal, I can type `bash launch.sh` and everything works as it should, I just can't get the desktop shortcut to run it.

**Reply by ltdrdata** [reply]
copy one of .desktop file in `/usr/share/applications` and edit it.

**Reply by mistamichal** [reply]
I tried that, copying the one I made for Automatic1111.
This is the Automatic1111 shortcut, which works:

```
[Desktop Entry]
Version=1.0
Name=Stable Diffusion
Comment=This runs Automatic1111
Path=/home/michael/sd-local/stable-diffusion-webui/
Exec=/home/michael/sd-local/stable-diffusion-webui/./launch.sh
Icon=/home/michael/sd-local/stable-diffusion-webui/icon.png
Terminal=true
Type=Application
Categories=Utility;Application;
```

and this is the one I copied and changed for Comfyui, which doesn't work:

```
[Desktop Entry]
Version=1.0
Name=ComfyUI
Comment=This runs ComfyUI
Path=/home/michael/ComfyUI/
Exec=/home/michael/ComfyUI/./launch.sh
Icon=/home/michael/ComfyUI/icon.png
Terminal=true
Type=Application
Categories=Utility;Application;
```
Although, Automatic1111 runs in a venv. Maybe I should put ComfyUI in a venv, too?

**Reply by nerdyrodent** [reply]
This is what I use:
```
[Desktop Entry]
Type=Application
Terminal=true
GenericName[en_GB]=ComfyUI
Name=ComfyUI
Comment=ComfyUI
Icon=utilities-terminal
Categories=Application; 
Exec=bash -ic "/home/nerdy/github/ComfyUI/nr_start.sh"
```
For my startup script I use:
```
#!/bin/bash
echo "Updating..."
cd /home/nerdy/github/ComfyUI
git pull
echo "Update completed. Starting ComfyUI."
conda run -n comfyui python /home/nerdy/github/ComfyUI/main.py --auto-launch
```

**Reply by mistamichal** [reply]
That did the trick. Thank you! :)
